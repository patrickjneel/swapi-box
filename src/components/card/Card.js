import React from 'react';
import './card.css';
import imperialEmblem from '../../imperialEmblem.svg';
import PropTypes from 'prop-types';

const Card = ( {itemData, location, updateFavorite, type} ) => {
  let mapped;
  if (location === 'planets') {
    if (itemData.residents.length) {
      mapped = itemData.residents.map(people => {
        return <div key={people}>{people}</div>;
      });
    } else {
      mapped = 0;
    }
  }
 
  return (
    <div className="card">
      <div className="top-card">
        <h3>{itemData.name}</h3>
        <img 
          className={type}
          onClick={() => updateFavorite(itemData)}
          src={imperialEmblem}/>
      </div>
      { 
        itemData.homeworld &&
        <div className="card-data">
          <h4>Planet:</h4> <h5>{itemData.homeworld}</h5>
          <h4>Population:</h4> <h5>{itemData.population}</h5>
          <h4>Species:</h4> <h5>{itemData.species}</h5>
        </div>
      }
      {
        itemData.climate &&
        <div className="card-data">
          <h4>Terrain:</h4> <h5>{itemData.terrrain}</h5>
          <h4>Climate:</h4> <h5>{itemData.climate}</h5>
          <h4>Residents:</h4> <h5>{mapped}</h5>
        </div>
      }
      {
        itemData.model &&
        <div className="card-data">
          <h4>Model:</h4> <h5>{itemData.model}</h5>
          <h4>Class:</h4> <h5>{itemData.class}</h5>
          <h4>Passengers:</h4><h5>{itemData.passengers}</h5>
          <h4>Manufacturer:</h4><h5>{itemData.manufacturer}</h5>
        </div>
      }
      
    </div>
  );
};

export default Card;

Card.propTypes = {
  itemData: PropTypes.object,
  location: PropTypes.string,
  updateFavorite: PropTypes.func,
  type: PropTypes.string
};
