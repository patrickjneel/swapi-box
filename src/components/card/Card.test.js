import React from 'react';
import { shallow, mount } from 'enzyme';
import Card from './Card';

describe('Card Test', () => {


  it('should be defined', () => {
const mockFunc = jest.fn();
    const mockData = [
      {
        homeworld: "Tatooine", 
        name: "Darth Vader", 
        species: "Human", 
        population: "200000"
     }
    ]
    const mockLocation = 'people';
    const mockFavs = [];
    const renderedCard = shallow(
      <Card
        itemData={mockData} 
        location={mockFunc} 
        updateFavorite={mockFunc} 
        favorites={mockFunc}
      />);

    expect(renderedCard).toBeDefined();
  });

   it('should receive props', () => {
const mockFunc = jest.fn();
    const mockData = 
      {
        homeworld: "Tatooine", 
        name: "Darth Vader", 
        species: "Human", 
        population: "200000"
     }
    
    const mockLocation = 'people';
    const mockFavs = [];
    const mockType= 'imperial'
    const renderedCard = mount(
      <Card
        itemData={mockData} 
        location={mockLocation} 
        updateFavorite={mockFunc} 
        favorites={mockFunc}
        type={mockType}
      />);
    expect(renderedCard.props().itemData).toEqual(mockData)
    expect(renderedCard.props().location).toEqual(mockLocation)
    expect(renderedCard.props().updateFavorite).toEqual(mockFunc)
    expect(renderedCard.props().favorites).toEqual(mockFunc)
    expect(renderedCard.props().type).toEqual(mockType)
  });


   it('should match snap shot', () => {
const mockFunc = jest.fn();
    const mockData = 
      {
        homeworld: "Tatooine", 
        name: "Darth Vader", 
        species: "Human", 
        population: "200000"
     }
    
    const mockLocation = 'people';
    const mockFavs = [];
    const type= 'imperial'
    const renderedCard = mount(
      <Card
        itemData={mockData} 
        location={mockLocation} 
        updateFavorite={mockFunc} 
        favorites={mockFunc}
        type={type}
      />);

    expect(renderedCard).toMatchSnapshot();

  });


});

