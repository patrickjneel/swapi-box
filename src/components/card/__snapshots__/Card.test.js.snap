// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Test should match snap shot 1`] = `
ReactWrapper {
  "length": 1,
  Symbol(enzyme.__unrendered__): <Card
    favorites={[Function]}
    itemData={
        Object {
            "homeworld": "Tatooine",
            "name": "Darth Vader",
            "population": "200000",
            "species": "Human",
          }
    }
    location="people"
    type="imperial"
    updateFavorite={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "favorites": [Function],
      "itemData": Object {
        "homeworld": "Tatooine",
        "name": "Darth Vader",
        "population": "200000",
        "species": "Human",
      },
      "location": "people",
      "type": "imperial",
      "updateFavorite": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": <div
        class="card"
>
        <div
                class="top-card"
        >
                <h3>
                        Darth Vader
                </h3>
                <img
                        class="imperial"
                        src="imperialEmblem.svg"
                />
        </div>
        <div
                class="card-data"
        >
                <h4>
                        Planet:
                </h4>
                <h5>
                        Tatooine
                </h5>
                <h4>
                        Population:
                </h4>
                <h5>
                        200000
                </h5>
                <h4>
                        Species:
                </h4>
                <h5>
                        Human
                </h5>
        </div>
</div>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="top-card"
>
            <h3>
                        Darth Vader
            </h3>
            <img
                        className="imperial"
                        onClick={[Function]}
                        src="imperialEmblem.svg"
            />
</div>,
          <div
            className="card-data"
>
            <h4>
                        Planet:
            </h4>
             
            <h5>
                        Tatooine
            </h5>
            <h4>
                        Population:
            </h4>
             
            <h5>
                        200000
            </h5>
            <h4>
                        Species:
            </h4>
             
            <h5>
                        Human
            </h5>
</div>,
          undefined,
          undefined,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <div
            class="top-card"
>
            <h3>
                        Darth Vader
            </h3>
            <img
                        class="imperial"
                        src="imperialEmblem.svg"
            />
</div>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                Darth Vader
</h3>,
              <img
                className="imperial"
                onClick={[Function]}
                src="imperialEmblem.svg"
/>,
            ],
            "className": "top-card",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <h3>
                Darth Vader
</h3>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Darth Vader",
              },
              "ref": null,
              "rendered": Array [
                "Darth Vader",
              ],
              "type": "h3",
            },
            Object {
              "instance": <img
                class="imperial"
                src="imperialEmblem.svg"
/>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "imperial",
                "onClick": [Function],
                "src": "imperialEmblem.svg",
              },
              "ref": null,
              "rendered": Array [
                undefined,
              ],
              "type": "img",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": <div
            class="card-data"
>
            <h4>
                        Planet:
            </h4>
            <h5>
                        Tatooine
            </h5>
            <h4>
                        Population:
            </h4>
            <h5>
                        200000
            </h5>
            <h4>
                        Species:
            </h4>
            <h5>
                        Human
            </h5>
</div>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4>
                Planet:
</h4>,
              " ",
              <h5>
                Tatooine
</h5>,
              <h4>
                Population:
</h4>,
              " ",
              <h5>
                200000
</h5>,
              <h4>
                Species:
</h4>,
              " ",
              <h5>
                Human
</h5>,
            ],
            "className": "card-data",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <h4>
                Planet:
</h4>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Planet:",
              },
              "ref": null,
              "rendered": Array [
                "Planet:",
              ],
              "type": "h4",
            },
            " ",
            Object {
              "instance": <h5>
                Tatooine
</h5>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Tatooine",
              },
              "ref": null,
              "rendered": Array [
                "Tatooine",
              ],
              "type": "h5",
            },
            Object {
              "instance": <h4>
                Population:
</h4>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Population:",
              },
              "ref": null,
              "rendered": Array [
                "Population:",
              ],
              "type": "h4",
            },
            " ",
            Object {
              "instance": <h5>
                200000
</h5>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "200000",
              },
              "ref": null,
              "rendered": Array [
                "200000",
              ],
              "type": "h5",
            },
            Object {
              "instance": <h4>
                Species:
</h4>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Species:",
              },
              "ref": null,
              "rendered": Array [
                "Species:",
              ],
              "type": "h4",
            },
            " ",
            Object {
              "instance": <h5>
                Human
</h5>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Human",
              },
              "ref": null,
              "rendered": Array [
                "Human",
              ],
              "type": "h5",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "favorites": [Function],
        "itemData": Object {
          "homeworld": "Tatooine",
          "name": "Darth Vader",
          "population": "200000",
          "species": "Human",
        },
        "location": "people",
        "type": "imperial",
        "updateFavorite": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": <div
          class="card"
>
          <div
                    class="top-card"
          >
                    <h3>
                              Darth Vader
                    </h3>
                    <img
                              class="imperial"
                              src="imperialEmblem.svg"
                    />
          </div>
          <div
                    class="card-data"
          >
                    <h4>
                              Planet:
                    </h4>
                    <h5>
                              Tatooine
                    </h5>
                    <h4>
                              Population:
                    </h4>
                    <h5>
                              200000
                    </h5>
                    <h4>
                              Species:
                    </h4>
                    <h5>
                              Human
                    </h5>
          </div>
</div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="top-card"
>
              <h3>
                            Darth Vader
              </h3>
              <img
                            className="imperial"
                            onClick={[Function]}
                            src="imperialEmblem.svg"
              />
</div>,
            <div
              className="card-data"
>
              <h4>
                            Planet:
              </h4>
               
              <h5>
                            Tatooine
              </h5>
              <h4>
                            Population:
              </h4>
               
              <h5>
                            200000
              </h5>
              <h4>
                            Species:
              </h4>
               
              <h5>
                            Human
              </h5>
</div>,
            undefined,
            undefined,
          ],
          "className": "card",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <div
              class="top-card"
>
              <h3>
                            Darth Vader
              </h3>
              <img
                            class="imperial"
                            src="imperialEmblem.svg"
              />
</div>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3>
                  Darth Vader
</h3>,
                <img
                  className="imperial"
                  onClick={[Function]}
                  src="imperialEmblem.svg"
/>,
              ],
              "className": "top-card",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": <h3>
                  Darth Vader
</h3>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Darth Vader",
                },
                "ref": null,
                "rendered": Array [
                  "Darth Vader",
                ],
                "type": "h3",
              },
              Object {
                "instance": <img
                  class="imperial"
                  src="imperialEmblem.svg"
/>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "imperial",
                  "onClick": [Function],
                  "src": "imperialEmblem.svg",
                },
                "ref": null,
                "rendered": Array [
                  undefined,
                ],
                "type": "img",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": <div
              class="card-data"
>
              <h4>
                            Planet:
              </h4>
              <h5>
                            Tatooine
              </h5>
              <h4>
                            Population:
              </h4>
              <h5>
                            200000
              </h5>
              <h4>
                            Species:
              </h4>
              <h5>
                            Human
              </h5>
</div>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h4>
                  Planet:
</h4>,
                " ",
                <h5>
                  Tatooine
</h5>,
                <h4>
                  Population:
</h4>,
                " ",
                <h5>
                  200000
</h5>,
                <h4>
                  Species:
</h4>,
                " ",
                <h5>
                  Human
</h5>,
              ],
              "className": "card-data",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": <h4>
                  Planet:
</h4>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Planet:",
                },
                "ref": null,
                "rendered": Array [
                  "Planet:",
                ],
                "type": "h4",
              },
              " ",
              Object {
                "instance": <h5>
                  Tatooine
</h5>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Tatooine",
                },
                "ref": null,
                "rendered": Array [
                  "Tatooine",
                ],
                "type": "h5",
              },
              Object {
                "instance": <h4>
                  Population:
</h4>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Population:",
                },
                "ref": null,
                "rendered": Array [
                  "Population:",
                ],
                "type": "h4",
              },
              " ",
              Object {
                "instance": <h5>
                  200000
</h5>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "200000",
                },
                "ref": null,
                "rendered": Array [
                  "200000",
                ],
                "type": "h5",
              },
              Object {
                "instance": <h4>
                  Species:
</h4>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Species:",
                },
                "ref": null,
                "rendered": Array [
                  "Species:",
                ],
                "type": "h4",
              },
              " ",
              Object {
                "instance": <h5>
                  Human
</h5>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Human",
                },
                "ref": null,
                "rendered": Array [
                  "Human",
                ],
                "type": "h5",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
